osu.Game/Overlays/BeatmapSet/PreviewButton.cs

```diff
@@ -82,7 +83,8 @@ protected override void Update()
 
             if (Playing.Value && preview != null)
             {
-                progress.Width = (float)(preview.CurrentTime / preview.Length);
+                // prevent negative (potential infinite) width if a track without length was loaded
+                progress.Width = (float)Math.Max(preview.CurrentTime / preview.Length, 0);

```

Comment:
not sure `Math.Max` is the best way of handling this. how about a check on `Length > 0` instead?

Reply:
Instead this?
```
// prevent negative (potential infinite) width if a track without length was loaded
if(preview.Length > 0)
  progress.Width = (float)preview.CurrentTime / preview.Length;
else
  progress.Width = 0;
```
May be a bit more verbose but also easier to read.

Reply:
can probably use tertiary ?: but yeah, that sounds fine.